nameservice/errors.go

```diff
@@ -26,8 +26,25 @@ var (
 	ErrNotFound        = errors.New("not found")
 	ErrInvalidName     = errors.New("invalid name")
 	ErrInvalidAddress  = errors.New("invalid address")
+
+	// RFC 1035 error interpretation
+	ErrNon      = errors.New("No Error")
```

Comment:
Its best to remove this as its unclear it no error is not an error. E.g. you'd have to do something like
```
if err!=nil && err != ErrNon{
....
}
```